api/v1alpha4/machineset_types.go

```diff
@@ -24,6 +24,12 @@ import (
 	capierrors "sigs.k8s.io/cluster-api/errors"
 )
 
+const (
+	// MachineSetTopologyFinalizer is the finalizer used by the topology MachineDeployment controller to
```

Comment:
Do these templates have owner refs? If so they would just go orphan and garbage collected so no need to explicitly take care of them?